cognite_toolkit/_cdf_tk/prototypes/commands/dump_timeseries.py

```diff
@@ -0,0 +1,306 @@
+                    if file_path.exists():
+                        with file_path.open("a", encoding=self.encoding, newline=self.newline) as f:
+                            f.write("\n")
+                            f.write(yaml.safe_dump(time_series, sort_keys=False))
+                    else:
+                        with file_path.open("w", encoding=self.encoding, newline=self.newline) as f:
+                            f.write(yaml.safe_dump(time_series, sort_keys=False))
+                    count += len(time_series)
+                    progress.advance(write_to_file, advance=len(time_series))
+            elif format_ in {"csv", "parquet"}:
+                file_count: int = 0  # Counter()
+                for df in self._buffer(writeable):
+                    folder_path = output_dir / TimeSeriesLoader.folder_name
+                    folder_path.mkdir(parents=True, exist_ok=True)
+                    file_path = folder_path / f"part-{file_count:04}.TimeSeries.{format_}"
+                    if format_ == "csv":
+                        df.to_csv(
+                            file_path,
+                            index=False,
+                            encoding=self.encoding,
+                            lineterminator=self.newline,
+                        )
+                    elif format_ == "parquet":
+                        df.to_parquet(file_path, index=False)
+                    file_count += 1
+                    if verbose:
+                        print(f"Dumped {len(df):,} time_series to {file_path}")
+                    count += len(df)
+                    progress.advance(write_to_file, advance=len(df))
+            else:
+                raise ToolkitValueError(f"Unsupported format {format_}. Supported formats are yaml, csv, parquet. ")
+
+        print(f"Dumped {count:,} time_series to {output_dir}")
+
+        if self._used_data_sets:
+            to_dump = DataSetWriteList(
+                [self.data_set_by_id[used_dataset] for used_dataset in self._used_data_sets]
+            ).dump_yaml()
+            file_path = output_dir / DataSetsLoader.folder_name / "time_series.DataSet.yaml"
+            file_path.parent.mkdir(parents=True, exist_ok=True)
+            if file_path.exists():
+                with file_path.open("a", encoding=self.encoding, newline=self.newline) as f:
+                    f.write("\n")
+                    f.write(to_dump)
+            else:
+                with file_path.open("w", encoding=self.encoding, newline=self.newline) as f:
+                    f.write(to_dump)
+
+            print(f"Dumped {len(self.data_set_by_id):,} data sets to {file_path}")
+
+    def _buffer(self, time_series_iterator: Iterator[list[dict[str, Any]]]) -> Iterator[pd.DataFrame]:
+        """Iterates over time_series until the buffer reaches the filesize."""
+        stored_time_series: pd.DataFrame = pd.DataFrame()
```

Comment:
Is `mypy` difficult on this one? Should not be necessary
```suggestion
        stored_time_series = pd.DataFrame()
```